src/esriSystem.ts

```diff
@@ -16,45 +16,84 @@ declare var require: Function;
 
 module esriSystem {
 
-  // return just the last part of a module name
-  function moduleName(name, overrides) {
-    if (overrides && overrides[name]) {
-      return overrides[name];
-    } else {
-      return name.match(/[^\/]+$/).shift();
+    // return just the last part of a module name
+    function moduleName(name: string, overrides?: any) {
+        if (overrides && overrides[name]) {
+            return overrides[name];
+        } else {
+            return name.match(/[^\/]+$/).shift();
+        }
+    }
+
+    //get the module name from the url that system js thinks it's loading it from
+    function getName(location: string) {
+        let startIndex = location.lastIndexOf("esri");
+        if (startIndex === -1) startIndex = location.lastIndexOf("dojo");
```

Comment:
Ultimately this probably needs to check for the other Dojo packages like 'dijit', 'dojox', etc, right? I'm still not totally clear how this works (I'll have to do some `console.log`ing so I can see what's going on, but the main thing is, it works!